entity-service-impl/src/main/java/org/hypertrace/entity/query/service/EntityQueryServiceImpl.java

```diff
@@ -367,6 +367,12 @@ public void update(EntityUpdateRequest request, StreamObserver<ResultSetChunk> r
 
   private void doUpdate(RequestContext requestContext, EntityUpdateRequest request)
       throws Exception {
+    Optional<String> maybeTenantId = requestContext.getTenantId();
+    if (maybeTenantId.isEmpty()) {
+      return;
```

Comment:
Will we throw an exception eventually as this one is private?

Reply:
The caller already checks the presence of tenant id in request context, and returns an error